integration-tests/bats/sql-shell.bats

```diff
@@ -42,61 +42,109 @@ teardown() {
     [[ "$output" =~ "+---------------------" ]] || false
 }
 
-@test "sql-shell: dolt sql shell has mysql db" {
-    # mysql database exists and has privilege tables
-    run dolt sql <<< "show tables from mysql;"
-    [ "$status" -eq 0 ]
-    [[ "$output" =~ "user" ]] || false
-    [[ "$output" =~ "role_edges" ]] || false
+@test "sql-shell: dolt sql shell with default doltcfg directory" {
+    # remove existing .doltcfg
+    rm -rf .doltcfg
 
     # show users, expect just root user
     run dolt sql <<< "select user from mysql.user;"
     [[ "$output" =~ "root" ]] || false
     ! [[ "$output" =~ "new_user" ]] || false
 
-    # create a new user, fails when no privilege file is specified
+    # check that .doltcfg exists
+    run ls -a
+    [[ "$output" =~ ".doltcfg" ]] || false
+
+    # check that ./doltcfg/privileges.db does not exist
+    run ls .doltcfg
+    ! [[ "$output" =~ "privileges.db" ]] || false
+
+    # create a new user
     run dolt sql <<< "create user new_user;"
-    [ "$status" -eq 1 ]
-    [[ "$output" =~ "no privilege file specified, to persist users/grants run with --privilege-file=<file_path>" ]] || false
+    [ "$status" -eq 0 ]
 
-    # there shouldn't be any mysql.db files
-    run ls .dolt
-    ! [[ "$output" =~ "mysql.db" ]] || false
+    # show users, expect root user and new_user
+    run dolt sql <<< "select user from mysql.user;"
+    [[ "$output" =~ "root" ]] || false
+    [[ "$output" =~ "new_user" ]] || false
+
+    # check that .doltcfg exists
+    run ls -a
+    [[ "$output" =~ ".doltcfg" ]] || false
+
+    # check that ./doltcfg/privileges.db exists
+    run ls .doltcfg
+    [[ "$output" =~ "privileges.db" ]] || false
 
     # remove mysql.db just in case
     rm -f .dolt/mysql.db
 }
 
-@test "sql-shell: dolt sql shell can create users" {
-    # remove existing privs.db
-    rm -f privs.db
+@test "sql-shell: dolt sql shell specify doltcfg directory" {
```

Comment:
Need a test that all of `doltcfg-dir`, `privileges-file` and `data-dir` can be set to totally different things and it all works